src/main/java/build/buildfarm/instance/memory/MemoryInstance.java

```diff
@@ -186,7 +186,8 @@ protected void updateOperationWatchers(Operation operation) {
   protected Operation createOperation(Action action) {
     String name = createOperationName(UUID.randomUUID().toString());
 
-    watchers.put(name, new ArrayList<Function<Operation, Boolean>>());
+    List<Function<Operation, Boolean>> previous = watchers.put(name, new ArrayList<Function<Operation, Boolean>>());
+    assert previous != null : "Operation name " + previous + " expected to be unique - but seems like there were some collisions";
```

Comment:
Assert is frowned upon by Google style, since it's usually not enabled - prefer Preconditions.checkState or Asserts.checkState (I think it's called Asserts - the class in guava).

This assert actually looks incorrect - don't you want to assert previous == null?